core/src/main/java/org/apache/calcite/sql/SqlLiteral.java

```diff
@@ -376,8 +378,13 @@ public <T extends Object> T getValueAs(Class<T> clazz) {
       final SqlIntervalLiteral.IntervalValue valTime =
           (SqlIntervalLiteral.IntervalValue) value;
       if (clazz == Long.class) {
-        return clazz.cast(valTime.getSign()
-            * SqlParserUtil.intervalToMillis(valTime));
+        if (typeName.getName().equals("INTERVAL_WEEK")) {
```

Comment:
Rather than make this change can `SqlParserUtil.intervalToMillis` just handle weeks as well?